### PR TITLE
[CARBONDATA-641]supported date type for dictionary_exclude

### PR DESCRIPTION
**Pull request by anubhav100**
DICTIONARY_EXCLUDE is not working with 'DATE' column

I am trying to create a table with "DICTIONARY_EXCLUDE" and this property is not working for "DATE" Data Type.
Query : CREATE TABLE uniqdata_date_dictionary (CUST_ID int,CUST_NAME string,ACTIVE_EMUI_VERSION string, DOB date, DOJ date, BIGINT_COLUMN1 bigint,BIGINT_COLUMN2 bigint,DECIMAL_COLUMN1 decimal(30,10), DECIMAL_COLUMN2 decimal(36,10),Double_COLUMN1 double, Double_COLUMN2 double,INTEGER_COLUMN1 int) STORED BY 'org.apache.carbondata.format' TBLPROPERTIES ("TABLE_BLOCKSIZE"= "256 MB","DICTIONARY_EXCLUDE"="DOB,DOJ");
Expected Result : Table created.
Actual Result : Error: org.apache.carbondata.spark.exception.MalformedCarbonCommandException: DICTIONARY_EXCLUDE is unsupported for date data type column: dob (state=,code=0)

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/624/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/625/

**Comment by ravipesala** [comment]
@anubhav100 Please add testcase in spark-common-test module

File: integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala [hunk 1]
```diff
@@ -590,7 +590,7 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
    * detects whether double or decimal column is part of dictionary_exclude
    */
   def isStringAndTimestampColDictionaryExclude(columnDataType: String): Boolean = {
-    val dataTypes = Array("string", "timestamp")
+    val dataTypes = Array("string", "timestamp", "date")
```
Comment (mohammadshahidkhan) [inline integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala]: @anubhav100 Please check my jira comment
this is wrong you should not allow adding timestamp and date in dictionary_exclude

Comment (mohammadshahidkhan) [inline integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala]: Please handle the same in TableCreator.isStringAndTimestampColDictionaryExclude

**Comment by anubhav100** [comment]
@ravipesala @mohammadshahidkhan changes are done

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/627/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/628/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/633/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/634/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/635/

**Comment by mohammadshahidkhan** [comment]
@anubhav100 
1. Please check you have missed my second comment TableCreator.isStringAndTimestampColDictionaryExclude 
2. Please give meaning full name to method isStringAndTimestampColDictionaryExclude in TableCreator and CarbonDDLSqlParser

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/636/

**Comment by anubhav100** [comment]
@mohammadshahidkhan can you have another look changes are done

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**